pkg/sql/catalog/nstree/map.go

```diff
@@ -34,10 +34,12 @@ type EntryIterator func(entry catalog.NameEntry) error
 
 // Upsert adds the descriptor to the tree. If any descriptor exists in the
 // tree with the same name or id, it will be removed.
-func (dt *Map) Upsert(d catalog.NameEntry) {
+func (dt *Map) Upsert(d catalog.NameEntry, skipNameMap bool) {
```

Comment:
That's a good good good catch. I totally missed that!
Yeah, storing functions by id separately is a good way. Also I'm thinking about extending the `NameEntry` interface to have the `SkipNameSpace` method instead of having it in the descriptor so that we can choose to delete or not by looking the `NameEntry` itself instead of passing the flag around. The downside is that having `SkipNameSpace` for a thing named `NameEntry` is super weird.... I'm inclined to your suggestion actually. But let me know.

Reply:
Another idea is to have a map tracking which id skipped that name map. I think this is better because the map data structure doesn't need to know about the concept of `Descriptors`. I'll give this a try shortly.

Reply:
Updated with an extra map tracking ids with name map skipped. I found this is a lot more straight forward than storing descriptors separately and of course a lazy change. Let me know how you feel about that.